src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -588,14 +581,21 @@ class CornerstoneViewport extends Component {
   };
 
   onNewImage = event => {
-    const newImageId = event.detail.image.imageId;
-    const newImageIdIndex = this.props.imageIds.indexOf(newImageId);
+    const { imageId } = event.detail.image;
+    const { sopInstanceUid } = cornerstone.metaData.get(
+      'generalImageModule',
+      imageId
+    );
+    const currentImageIdIndex = this.props.imageIds.indexOf(imageId);
 
     // TODO: Should we grab and set some imageId specific metadata here?
     // Could prevent cornerstone dependencies in child components.
-    this.setState({
-      imageIdIndex: newImageIdIndex,
-    });
+    if (this.props.setViewportSpecificData) {
+      this.props.setViewportSpecificData({
+        currentImageIdIndex,
+        sopInstanceUid,
```

Comment:
What do we use the `sopInstanceUid` for here?

Reply:
`cornerstone` extension uses sopInstanceUid to decide if its state should be updated.